cmake/patches/dawn/dawn.patch

```diff
@@ -79,3 +79,111 @@ index 0037d83276..6372c4ee77 100644
    tint_lang_wgsl_program
    tint_lang_wgsl_sem
    tint_lang_wgsl_writer_ir_to_program
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index bcb91ba106..4e54940c50 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -188,6 +188,8 @@ option(DAWN_ENABLE_PIC "Build with Position-Independent-Code enabled" OFF)
+ option(DAWN_EMIT_COVERAGE "Emit code coverage information" OFF)
+ set_if_not_defined(LLVM_SOURCE_DIR "${Dawn_LLVM_SOURCE_DIR}" "Directory to an LLVM source checkout. Required to build turbo-cov")
+
++option(DAWN_CROSSCOMPILING_WIN "Support windows platform crosscompiling, mainly for build arm/aarch64 target on x64 platform" OFF)
++
+ message(STATUS "Dawn build with asserts in all configurations: ${DAWN_ALWAYS_ASSERT}")
+ message(STATUS "Dawn build Wayland support: ${DAWN_USE_WAYLAND}")
+ message(STATUS "Dawn build X11 support: ${DAWN_USE_X11}")
+@@ -377,6 +380,11 @@ target_include_directories(dawn_internal_config INTERFACE
+ )
+ target_link_libraries(dawn_internal_config INTERFACE dawn_public_config)
+
++# Disable CMAKE_CROSSCOMPING because we don't have full support
++if(CMAKE_CROSSCOMPILING)
++  set(CMAKE_CROSSCOMPILING OFF)
```

Comment:
But we are doing cross-compiling? Usually the var is set to ON when the [CMAKE_SYSTEM_NAME](https://cmake.org/cmake/help/latest/variable/CMAKE_SYSTEM_NAME.html#variable:CMAKE_SYSTEM_NAME) variable has been set manually.  And usually we only read from this var, we do not set it. I worry a lot of other cmake functions/modules may rely on this var.

Reply:
I think we can remove this manually switch. Let me have a try locally.